npm/binary_install.js

```diff
@@ -0,0 +1,113 @@
+const { existsSync, mkdirSync, createWriteStream } = require("fs");
+const { join } = require("path");
+const { spawnSync } = require("child_process");
+
+const axios = require("axios");
+
+const error = (msg) => {
+  console.error(msg);
+  process.exit(1);
+};
+
+class Binary {
+  constructor(name, url) {
+    let errors = [];
+    if (typeof url !== "string") {
+      errors.push("url must be a string");
+    } else {
+      try {
+        new URL(url);
+      } catch (e) {
+        errors.push(e);
+      }
+    }
+    if (name && typeof name !== "string") {
+      errors.push("name must be a string");
+    }
+
+    if (!name) {
+      errors.push("You must specify the name of your binary");
+    }
+    if (errors.length > 0) {
+      let errorMsg =
+        "One or more of the parameters you passed to the Binary constructor are invalid:\n";
+      errors.forEach((error) => {
+        errorMsg += error;
+      });
+      errorMsg +=
+        '\n\nCorrect usage: new Binary("my-binary", "https://example.com/binary/download.tar.gz")';
+      error(errorMsg);
+    }
+    this.url = url;
+    this.name = name;
+    this.installDirectory = join(__dirname, "node_modules", ".bin");
+
+    if (!existsSync(this.installDirectory)) {
+      mkdirSync(this.installDirectory, { recursive: true });
+    }
+
+    this.binaryPath = join(this.installDirectory, this.name);
+  }
+
+  exists() {
+    return existsSync(this.binaryPath);
+  }
+
+  install(fetchoptions, suppressLogs = false) {
+    if (this.exists()) {
+      if (!suppressLogs) {
+        console.error(
+          `${this.name} is already installed, skipping installation.`
+        );
+      }
+      return Promise.resolve();
+    }
+
+    if (existsSync(this.installDirectory)) {
+      rimraf.sync(this.installDirectory);
```

Comment:
First of, you do not required rimraf. Second of, can we use native node `fs` api here?

Reply:
Okkk I'll write it myself